src/Http/Controllers/ActionController.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+namespace HeadlessLaravel\Formations\Http\Controllers;
+
+use Illuminate\Bus\Batch;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Bus;
+use Throwable;
+
+class ActionController extends Controller
+{
+    public function store(Request $request)
+    {
+        $currentAction = $this->formationAction();
+
+        if (!$currentAction) {
+            return redirect()->route('formation.index');
+        }
+
+        $modelIds = $request->get('selected', []);
+
+        $models = [];
+        if ($modelIds === 'all') {
+            $models = $currentAction->formation->results();
```

Comment:
I think this might not get `all` since results are always paginated

May have to call `->builder()->get()` to avoid that.

And maybe we could use [cursor()](https://laravel.com/docs/8.x/eloquent#cursors) to avoid memory issues for large sets

Reply:
Im also wondering how this works filters 🤔 

Like if they select all in a filtered set, would we currently support that?

If not we would have to figure out how to pass the current query string 

And how to handle it in the formation, maybe `formstion->defaults = $request->input('query')`

```php
'selected' => 'all',
'fields' => ['status' => 'active']
'query' => ['search' => 'laravel', 'sort-desc' => 'title', 'author' => 1]
```

Reply:
Let me check on this and come back to you.

Reply:
ok no problem. just brainstorming but maybe its another option for selected

```php
'selected' => ['search' => 'laravel', 'sort-desc' => 'title', 'author' => 1],
'fields' => ['status' => 'active']
```

if array && is associative, assume its filter parameters

Reply:
could probably add a method similar to applyDefaults that lets us:

```php
$this->formation()->given($params)->builder()->cursor()
```

`given` accepts params like defaults and applies them to request
`builder` exists and would handle the filtering
`cursor` would handle as a lazy collection

Reply:
I've used a builder to make a query and then apply the model id selection. Please check the latest code and let me know if that works.

I've added test and filters and sorting is now working with that.

Reply:
@mitulgolakiya looks good, I think the query parameters should be in `selected` or its own key `query`

```php
'selected' => 'all', // every model "select all"
'fields' => ['status' => 'active']
```

```php
'selected' => ['search' => 'laravel', 'sort-desc' => 'title', 'author' => 1], // "select all while filtered"
'fields' => ['status' => 'active']
```
Or maybe as a `query` which do you think is better?
```php
'selected' => 'all',
'query' => ['search' => 'laravel', 'sort-desc' => 'title', 'author' => 1], // alternative
'fields' => ['status' => 'active']
```

Right now I think its combined which might be tricky to deal with on the frontend

```php
'selected' => 'all',
''search' => 'laravel', 
'sort-desc' => 'title',
 'author' => 1,
'fields' => ['status' => 'active']
```

Reply:
I think `query` might be a good option.. thoughts?

Reply:
Done